src/content/docs/en/basics/astro-pages.mdx

```diff
@@ -96,10 +96,56 @@ Files with the `.html` file extension can be placed in the `src/pages/` director
 
 ## Custom 404 Error Page
 
-For a custom 404 error page, you can create a `404.astro` or `404.md` file in `/src/pages`.
+For a custom 404 error page, you can create a `404.astro` or `404.md` file in `src/pages`.
 
 This will build to a `404.html` page. Most [deploy services](/en/guides/deploy/) will find and use it.
 
+## Custom 500 Error Page
+
+For a custom 500 error page, you can create a `500.astro` in `src/pages`.
+
+:::caution
+Once deployed, Astro will only use `500.astro` when using `output: 'hybrid'` for pages that opt-out from prerendering or `output: 'server'`.
+:::
+
+If an error occurs in this file, it will default to the host 500 error page.
+
+### Retrieve The Error
+
+<p><Since v="4.10.0" /></p>
+
+`500.astro` accepts an `error` prop. It can be anything, so make sure to handle it properly:
```

Comment:
I find it weird to say that the "page accepts an error prop" when we don't show passing/giving a prop. 

Is this just something that is automatically available to this page, no work required? Or, do you create an error message and somewhere pass it to this page? Can someone literally just write what you've shown below and an error message is shown? (pulled from the 500 error message from the browser?) Or do you have to write an error message (or the logic to grab one) and then somehow pass it (and if so, how?) 

I think explaining or showing where the error comes from is important here.

Reply:
Yeah my example should work as is! Basically any error thrown during rendering (so that could be in a middleware, in an astro file or really anywhere) will end up there. There's no way to know what the error will be (you can throw strings for example, or even nothing) so it's really up to the user to handle that part.

Maybe we could rephrase to "accepts a special `error` prop"?

Reply:
Great, I can work with this and will suggest something! I just wanted to make sure I knew exactly how it worked first.